marlowe-tutorial/doc/marlowe-model.adoc

```diff
@@ -17,24 +17,61 @@ The Marlowe model allows for a contract to control money in a number of disjoint
 
 The mechanisms by which these blocks are generated, and by whom, are not relevant here, but contracts will be expressed in terms of _slot numbers_, counting from the starting (“genesis”) block of the blockchain.
 
 ==== UTxO and wallets
 
-Value on the blockchain resides in the UTxO, which are protected cryptographically by a private key held by the owner. These keys can be used to _redeem_ the output, and so to use them as inputs to new transactions, which can be seen as spending the value in the inputs. Users typically keep track of their private keys, and the values attached to them, in a cryptographically-secure _wallet_. 
+Value on the blockchain resides in the UTxO, which are protected cryptographically by a private key held by the owner. These keys can be used to _redeem_ the output, and so to use them as inputs to new transactions, which can be seen as spending the value in the inputs. Users typically keep track of their private keys, and the values attached to them, in a cryptographically-secure _wallet_.
 
 It is through their wallets that users are able to interact with smart contracts - including Marlowe contracts – running on the blockchain. Deposits are made from users’ wallets, and payments received by them. Note, however, that these are definitely _off chain_ actions that need to be controlled by code running in the user’s wallet: they cannot be made to happen by the Marlowe contract itself.
 
+==== Value
+
+In previous examples, whenever a `Value` was required, we have exclusively used Ada.
+This makes a lot of sense, seeing as Ada is the fundamental currency supported by Cardano.
+
+Plutus offers a more general concept of _value_, though,
+supporting custom fungible, non-fungible, and mixed tokens
+
+What is a `Value` in Plutus?
+
+[source,haskell]
+----
+newtype Value = Value
+    {getValue :: Map CurrencySymbol (Map TokenName Integer)}
+----
+
+Types `CurrencySymbol` and `TokenName` are both simple wrappers around
+`ByteString`.
+
+This notion of _value_ encompasses Ada, fungible tokens (think currencies),
+non-fungible tokens (a custom token that is not interchangeable with other tokens), and
+more exotic mixed cases:
+
+* Ada has the _empty bytestring_ as `CurrencySymbol` and `TokenName`.
+* A _fungible_ token is represented by a `CurrencySymbol` for which there is
+  exactly one `TokenSymbol` which can have an arbitrary non-negative quantity (of which Ada is a
+  special case).
+* A class of _non-fungible_ tokens is a `CurrencySymbol` with several ``TokenName``s,
+  each of which has a quantity of one. Each of these names corresponds to one
+  unique non-fungible token.
+* Mixed tokens are those with several ``TokenName``s _and_ quantities greater
+  than one.
+
+Cardano and Plutus provide a simple way to introduce a new currency by _forging_ it using
```

Comment:
Can we provide a cross-reference to the Plutus tutorial here?

Reply:
Apparently, there is nothing to reference in the Plutus tutorial yet.